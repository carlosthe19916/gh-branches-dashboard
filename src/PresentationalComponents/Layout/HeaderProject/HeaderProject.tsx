import * as React from "react";
import {
  PageHeader,
  Brand,
  Toolbar,
  ToolbarGroup,
  ToolbarItem
} from "@patternfly/react-core";
import { css } from "@patternfly/react-styles";
import accessibleStyles from "@patternfly/react-styles/css/utilities/Accessibility/accessibility";

import navBrandImage from "../../../brand.svg";
// import imgAvatar from "../../../avatar.svg";
import { PageNav } from "../PageNav";
import { SignInButton } from "../../SignInButton";

export interface HeaderProjectProps {
  aboutButton: React.ReactNode;
}

interface State {}

export class HeaderProject extends React.Component<HeaderProjectProps, State> {
  renderPageToolbar = () => {
    const { aboutButton } = this.props;
    return (
      <React.Fragment>
        <Toolbar>
          <ToolbarGroup
            className={css(
              accessibleStyles.screenReader,
              accessibleStyles.visibleOnLg
            )}
          >
            <ToolbarItem>{aboutButton}</ToolbarItem>
          </ToolbarGroup>
          <ToolbarGroup>
            <ToolbarItem>
              <SignInButton />
            </ToolbarItem>
          </ToolbarGroup>
        </Toolbar>
      </React.Fragment>
    );
  };

  render() {
    return (
      <PageHeader
        logo={<Brand src={navBrandImage} alt="gh-branches-dashboard" />}
        toolbar={this.renderPageToolbar()}
        // avatar={<Avatar src={imgAvatar} alt="Avatar image" />}
        topNav={<PageNav />}
      />
    );
  }
}
