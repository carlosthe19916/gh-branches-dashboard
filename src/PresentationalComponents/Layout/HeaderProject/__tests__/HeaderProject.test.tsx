import React from "react";
import { shallow } from "enzyme";
import { HeaderProject, HeaderProjectProps } from "../HeaderProject";
import { Button } from "@patternfly/react-core";

it("Test snapshot", () => {
  const props: HeaderProjectProps = {
    aboutButton: <Button>About Button</Button>
  };

  const wrapper = shallow(<HeaderProject {...props} />);
  expect(wrapper).toMatchSnapshot();
});
