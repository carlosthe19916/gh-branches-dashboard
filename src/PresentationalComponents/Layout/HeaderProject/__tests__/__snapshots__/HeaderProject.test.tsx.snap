// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
<PageHeader
  logo={
    <Brand
      alt="gh-branches-dashboard"
      src="brand.svg"
    />
  }
  toolbar={
    <React.Fragment>
      <Toolbar>
        <ToolbarGroup
          className="pf-u-screen-reader pf-u-visible-on-lg"
        >
          <ToolbarItem>
            <Unknown>
              About Button
            </Unknown>
          </ToolbarItem>
        </ToolbarGroup>
        <ToolbarGroup>
          <ToolbarItem>
            <SignInButton />
          </ToolbarItem>
        </ToolbarGroup>
      </Toolbar>
    </React.Fragment>
  }
  topNav={<PageNav />}
/>
`;
